onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -190,9 +190,22 @@ Status Conv<T>::UpdateState(OpKernelContext* context, bool bias_expected) const
     if (rank < 2) {
       // TODO: Explore padding the provided input shape [N, C, D] to [N, C, 1, D]
       // especially for EXHAUSTIVE algo search which may result in a better algo selection.
-      // Currently, we are padding it to [N, C, D, 1] as this seems to be the sweet spot
-      // for all algo search options: EXHAUSTIVE, HEURISTIC, and DEFAULT.
-      // See PR #7348 for more context.
+      // ORTModule uses different algo search options (HEURISTIC, and use max workspace size) compared to
+      // inference build (EXHAUSTIVE, 32M workspace size). We observed better perf when we pad input shape
+      // [N,C,D] to [N,C,1,D], expecially on A100, and excpecially for ConvGrad.
+      // PyTorch also pads to [N,C,1,D]. For inference build, we still pad it to [N, C, D, 1] as this seems
+      // to be the sweet spot for all algo search options: EXHAUSTIVE, HEURISTIC, and DEFAULT.
+      // See PR #7348 and #7702 for more context.
+#ifdef ENABLE_TRAINING
```

Comment:
How about introducing a provider option to allow the users to choose how they would like to deal with 1D conv ? When Sherlock introduced this logic about a year ago, we could see gains for some models during inferencing as well, but it caused a sever regression in one partner model and that was the reason it ultimately got reverted. With a provider option, the user gets to choose how they want to go about it. On the flip side, it is a detail of CUDNN that has to be exposed as a provider option (Not sure if that is a clean approach either)

Reply:
Right. I added the option, and keep it off by default and turn on it for ORTModule.